addon/components/file-browser/component.js

```diff
@@ -373,6 +398,107 @@ export default Ember.Component.extend(Analytics, {
         },
         dismissPop() {
             this.set('popupOpen', false);
-        }
+        },
+        checkNodeTitleKeypress(value) {
+            this.set('nodeTitle', value);
+            if (this.get('nodeTitle')) {
+                this.set('isInputInvalid', false);
+            } else {
+                this.set('isInputInvalid', true);
+            }
+        },
+        changeProjectSelectState(state) {
+            this.set('projectSelectState', state);
+            if (this.get('convertOrCopy')) {
+                this.set('convertOrCopy', null);
+            }
+            if (!this.get('isInputInvalid')) {
+                this.set('isInputInvalid', true);
+            }
+        },
+        updateConvertOrCopy(state) {
+            this.set('convertOrCopy', state);
+            if (this.get('convertOrCopy') == 'convert') {
+                this.set('isInputInvalid', false);
+            } else {
+                this.set('isInputInvalid', true);
+            }
+        },
+        setSelectedNode(node) {
+            this.set('node', node);
+            this.set('nodeLink', node.get('links.html'));
+        },
+        setMoveFile() {
+            if (this.get('projectSelectState') == 'newProject') {
+                this.send('createProject');
+            } else if (this.get('projectSelectState') == 'existingProject') {
+                if (this.get('convertOrCopy') == 'copy') {
+                    this.send('createComponent');
+                } else if (this.get('convertOrCopy') == 'convert') {
+                    this.send('moveToComponent');
+                }
+            }
+        },
```

Comment:
If we do the refactors mentioned below, we could reactor this action to be something more like:
```
if (this.get('projectSelectState') == 'newProject') {
    return this._createProject().then(project => {
        this.set('node', project);
        this.set('newNode', 'project');
        return this._moveFile(item, project);
    });
} else if (this.get('projectSelectState') == 'existingProject') {
.
.
.
```